Osteppy/src/server.js

```diff
@@ -1,32 +1,97 @@
 #!/usr/bin/env node
 
-/***********************************************************
+/** ********************************************************
 // OSTEP Dashboard Osteppy API
 // server.cpp
 // Date Created: 2018/09/11
 // Author: Olga Belavina and Yiran Zhu
 // Email: yzhu132@myseneca.ca
-// Description: Slack API triggered by and responds to 
-// slash commands such as /eod and /eod_left. 
-***********************************************************/
+// Description: Slack API triggered by and responds to
+// slash commands such as /eod and /eod_left.
+********************************************************* */
 
+import cp from 'child_process';
 import http from 'http';
+import path from 'path';
 import express from 'express';
 import cors from 'cors';
 import morgan from 'morgan';
 import bodyParser from 'body-parser';
-import config from './config.json';
 import axios from 'axios';
-import * as fs from 'fs';
+import fs from 'fs';
+import config from './config.json';
+
+class EODList {
+  constructor({
+    dataFile = path.join(__dirname, 'eods.json'),
+    unsubmittedNamesFile = path.join(__dirname, 'sleepyRAs.txt'),
+  } = {}) {
+    this.reportData = {};
+    this.dataFile = dataFile;
+    this.unsubmittedNames = [];
+    this.unsubmittedNamesFile = unsubmittedNamesFile;
+  }
+
+  async load() {
+    const results = await Promise.all([
+      fs.promises.readFile(this.dataFile, { encoding: 'utf8', flag: 'a+' }),
+      fs.promises.readFile(this.unsubmittedNamesFile, { encoding: 'utf8', flag: 'a+' }),
+    ]);
+    const [reportData, unsubmittedNames] = results;
+
+    try {
+      this.reportData = JSON.parse(reportData);
+    } catch (e) {
+      console.log(`Error parsing ${this.dataFile}: ${e}`);
+    }
+
+    this.unsubmittedNames = unsubmittedNames.split('\n');
+  }
+
+  save() {
+    return Promise.all([
+      fs.promises.writeFile(this.dataFile, JSON.stringify(this.reportData)),
+      fs.promises.writeFile(this.unsubmittedNamesFile, this.unsubmittedNames.join('\n')),
+    ]);
+  }
+
+  submit(name, data) {
+    this.reportData[name] = data;
+    this.unsubmittedNames = this.unsubmittedNames.filter(n => n !== name);
+    return this.save();
+  }
+
+  missing() {
+    return [...this.unsubmittedNames];
+  }
+
+  report() {
+    return { ...this.reportData };
+  }
```

Comment:
Just curious, why are you spreading in these 2 cases?

Reply:
My initial thinking was I wanted callers to avoid accidental data mutations, but since this is javascript and the data is accessible via `this`, it's probably unnecessary.